pulumi/integration_tests/__main__.py

```diff
@@ -0,0 +1,121 @@
+import os
+import sys
+from pathlib import Path
+
+sys.path.insert(0, "..")
+
+import pulumi_aws as aws
+import pulumi_nomad as nomad
+from infra import config
+from infra.autotag import register_auto_tags
+from infra.nomad_job import NomadJob, NomadVars
+from infra.quiet_docker_build_output import quiet_docker_output
+
+import pulumi
+
+
+def main() -> None:
+    ##### Preamble
+
+    stack_name = stackname_sans_prefix()
+
+    quiet_docker_output()
+
+    # These tags will be added to all provisioned infrastructure
+    # objects.
+    register_auto_tags({"grapl deployment": stack_name})
+
+    if not config.LOCAL_GRAPL:
+        # Set the nomad address. This can be either set as nomad:address in the config to support ssm port forwarding or
+        # taken from the nomad stack
+        nomad_config = pulumi.Config("nomad")
+        nomad_override_address = nomad_config.get("address")
+        # We prefer nomad:address to support overriding in the case of ssm port forwarding
+        nomad_server_stack = pulumi.StackReference(f"grapl/nomad/{stack_name}")
+        nomad_address = nomad_override_address or nomad_server_stack.require_output(
+            "address"
+        )
+        nomad_provider = nomad.Provider("nomad-aws", address=nomad_address)
+
+    ##### Actual Logic
+
+    grapl_stack = GraplStack(stack_name)
+
+    assert aws.config.access_key, "Appease typechecker"
+    assert aws.config.secret_key, "Appease typechecker"
```

Comment:
In the grapl pulumi the assert aws.config.access_key is to ensure that AWS keys are set. I'm pretty sure that this will break when compared to a specific string

Reply:
So, this logic actually changes in my upcoming PR (this works fine locally and breaks against AWS; this is what that whole `optionally-defined-env-vars` song and dance you did a while back is about)

Reply:
ah right, ok.